targets/simple_switch/simple_switch.cpp

```diff
@@ -267,6 +269,16 @@ SimpleSwitch::get_time_since_epoch_us() const {
   return duration_cast<ts_res>(tp.time_since_epoch()).count();
 }
 
+void
+SimpleSwitch::periodic_thread() {
+  auto actions = bm::PeriodicActionsList::get_instance();
+  while (!exiting) {
+    if (actions->next() != true) {
```

Comment:
It seems that if there is no periodic action registered, the thread exits right away. But what if I load a new P4 program that uses an extern with a periodic action?